analyzers/src/SonarAnalyzer.CSharp/Rules/BooleanLiteralUnnecessary.cs

```diff
@@ -99,5 +136,11 @@ private void CheckConditional(SonarSyntaxNodeReportingContext context)
 
         private static bool IsThrowExpression(ExpressionSyntax expressionSyntax) =>
             ThrowExpressionSyntaxWrapper.IsInstance(expressionSyntax);
+
+        private SyntaxNode GetIsPatternLeft(IsPatternExpressionSyntaxWrapper patternWrapper) =>
```

Comment:
This is basically what the removed `protected abstract SyntaxNode GetLeftNode(TBinaryExpression binary);` was for. You can treat 
* `left is right` like `left == right` and
* `left is not right` like `left != right`

On both sides you find expressions:

*  left is always an expression
* right in the `is right` case is `((ConstantPatternSyntaxWrapper)patternWrapper.Pattern).Expression` 
* right in the `is not right` is something like `((patternWrapper.Pattern as UnaryPattern).Pattern as ConstantPattern).Expression`

Reply:
```suggestion
        private SyntaxNode GetIsPatternLeft(IsPatternExpressionSyntaxWrapper isPattern) =>
```